src/Core/Services/RequestValidator.cs

```diff
@@ -154,7 +154,7 @@ public static void ValidateStoredProcedureRequest(string? primaryKeyRoute)
             // If query string or body contains extra parameters that don't exist
             // TO DO: If the request header contains x-ms-must-match custom header with value of "ignore"
             // this should not throw any error. Tracked by issue #158.
-            if (extraFields.Count > 0)
+            if (extraFields.Count > 0 && IsRequestBodyStrict())
```

Comment:
If request body is not strict, how do we handle extra fields? is the codepath the same for removing ("ignoring") those fields? can you remind me if you remove them from the request context somehow?

Reply:
Extra fields are not removed from request context since all the validations are performed on request context itself in the `RequestValidator `class. 

For mutations on tables/views, the extra fields are ignored when mutation parameters are created via `SqlMutationEngine.PrepareParameters()` method.

For stored procedures, this here:
https://github.com/Azure/data-api-builder/blob/402e86768a45762ed4a74e8b369ee5bf1f30daa1/src/Core/Resolvers/Sql%20Query%20Structures/SqlExecuteQueryStructure.cs#L39

processes params from the SP's definition. For all the params from the SP's definition, if the param is present in the request body, then only it is processed. Meaning, if there are any extra params specified in the request body, they are just ignored.

Reply:
Great, thanks for the context :)